src/Raven.Server/Utils/CertificateUtils.cs

```diff
@@ -13,17 +13,85 @@
 using Org.BouncyCastle.Security;
 using Org.BouncyCastle.X509;
 using Org.BouncyCastle.X509.Extension;
-using Raven.Server.ServerWide;
+using Raven.Server.Config.Categories;
 using Sparrow;
+using Sparrow.Logging;
 using Sparrow.Platform;
 using BigInteger = Org.BouncyCastle.Math.BigInteger;
 using X509Certificate = Org.BouncyCastle.X509.X509Certificate;
 
 namespace Raven.Server.Utils
 {
-    internal class CertificateUtils
+    internal static class CertificateUtils
     {
-        private const int BitsPerByte = 8; 
+        private const int BitsPerByte = 8;
+
+        private static readonly Logger Logger = LoggingSource.Instance.GetLogger("Server", typeof(CertificateUtils).FullName);
+
+        internal static bool CertHasKnownIssuer(X509Certificate2 userCertificate, X509Certificate2 knownCertificate, SecurityConfiguration securityConfiguration, out string issuerPinningHash)
+        {
+            issuerPinningHash = null;
+            X509Certificate2 issuerCertificate = null;
+
+            var userChain = new X509Chain();
+            var knownCertChain = new X509Chain();
```

Comment:
I think we need a comment here explaining that we explicitly don't set DisableCertificateDownloads